### ISSUE TITLE
User defined options with label & value in typedInput is possible but will not be handled correctly

### ISSUE THREAD

**Issue by volkerrichert** [node-red/node-red#1549]
If you create an typedInput with an user defined type, which has with more that an array of strings as options, you are not able to select it.

It can be done by defining a types like this and can be used to hide the real values from the user and also enable localisation:
`
              types: [ 'bool',
                    {
                        value: 'ir',
                        label: 'IR Commands',
                        options: [
                            {
                                value: 'POW',
                                label: 'Power Toggle'
                            }, {
                                value: 'SVM',
                                label: 'Set Verbose Mode'
                            }
                        ]
                    }
                ]
`

This will display the correct DropDown, because you set up this kind of object internally if a sting is provided only (https://github.com/node-red/node-red/blob/master/editor/js/ui/common/typedInput.js#L216).

### What are the steps to reproduce?

Create an typedInput with like above.

### What happens?

You can select an option correctly labeled, but in the callbacks only the value will be handled (https://github.com/node-red/node-red/blob/master/editor/js/ui/common/typedInput.js#L231) and validated against a simple array of Strings (https://github.com/node-red/node-red/blob/master/editor/js/ui/common/typedInput.js#L357). The result is an empty value.

### What do you expect to happen?
I expect the input field is set to the value of the selected object.

### Please tell us about your environment:

- [ ] Node-RED version: 0.17.5
- [ ] node.js version: 8.9.1
- [ ] npm version: 5.5.1
- [ ] Platform/OS: Win10
- [ ] Browser: Chrome

**Comment by volkerrichert** [comment]
Ok, here is a test node to show what I mean. There is a typedInput using the predefined msg type and add 2 userdefined. 'ir' has a combi of a sting and an object, like the one created in _createMenu. In this case the menu look fine and you can select both entries. 'test' works like expected, but 'POW' will be forced to '' beacise it'll not pass validation.

The second type 'set' just display [object Object].


```javascript
<script type="text/x-red" data-template-name="testnode">
    <div class="form-row">
        <label for="node-input-name"><i class="fa fa-tag"></i> Name</label>
        <input type="text" id="node-input-name" placeholder="Name">
    </div>

    </div>
        <div class="form-row">
        <label for="node-input-test"><i class="fa fa-globe"></i> test</label>
        <input type="text" id="node-input-test" style="width:70%">
        <input type="hidden" id="node-input-testType">
    </div>
</script>

<script type="text/x-red" data-help-name="testnode">
    <p>test node for user defined typed input with key and values</p>
</script>

<script type="text/javascript">
    RED.nodes.registerType('testnode', {
        category: 'output',
        color: '#006738',
        defaults: {
            name: {value: ""},
            test: { value:''},
            testType: { value: 'msg'}
        },
        inputs: 1,
        outputs: 0,
        align: 'right',
        label: function () {
            return (this.name || 'testnode');
        },
        oneditprepare: function () {
            $('#node-input-test').typedInput({
                default: "msg",
                typeField: $("#node-input-testType"),
                types: [ 'msg',
                    {
                        value: 'ir',
                        label: 'IR Commands',
                        options: [
                            'test',
                            {
                                value: 'POW',
                                label: 'Power Toggle'
                            }
                        ]
                    },
                    {
                        value: 'set',
                        label: 'Adv Commands',
                        options: [
                            {
                                value: 'SVM',
                                label: 'Set Verbose Mode'
                            }
                        ]
                    }
                ]
            });
        }
    });

</script>
```

**Comment by knolleary** [comment]
Fixed in the 0.18 branch.

**Comment by volkerrichert** [comment]
Cool, THX!
